Calibration/HcalAlCaRecoProducers/plugins/AlCaIsoTracksFilter.cc

```diff
@@ -0,0 +1,353 @@
+#include "Geometry/CaloTopology/interface/CaloSubdetectorTopology.h"
+#include "Geometry/CaloTopology/interface/HcalTopology.h"
+#include "Geometry/CaloTopology/interface/CaloTopology.h"
+#include "CondFormats/DataRecord/interface/EcalChannelStatusRcd.h"
+#include "Geometry/CaloTopology/interface/EcalTrigTowerConstituentsMap.h"
+#include "RecoLocalCalo/EcalRecAlgos/interface/EcalSeverityLevelAlgo.h"
+#include "RecoLocalCalo/EcalRecAlgos/interface/EcalSeverityLevelAlgoRcd.h"
+
+//
+// class declaration
+//
+
+class AlCaIsoTracksFilter : public edm::EDFilter {
+public:
+  explicit AlCaIsoTracksFilter(const edm::ParameterSet&);
+  ~AlCaIsoTracksFilter();
+  
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  
+private:
+  virtual bool filter(edm::Event&, const edm::EventSetup&) override;
+  virtual void endJob() override;
+  virtual void beginRun(edm::Run const&, edm::EventSetup const&);
+  virtual void endRun(edm::Run const&, edm::EventSetup const&);
+  
+  // ----------member data ---------------------------
+  HLTConfigProvider          hltConfig_;
+  std::vector<std::string>   trigNames, HLTNames;
+  std::vector<int>           trigKount, trigPass;
+  spr::trackSelectionParameters selectionParameters;
+  std::string                theTrackQuality, processName;
+  std::string                l1Filter, l2Filter, l3Filter;
+  double                     a_mipR, a_coneR, a_charIsoR;
+  double                     pTrackMin_, eEcalMax_, eIsolation_;
+  int                        nRun, nAll, nGood;
+  edm::InputTag              triggerEvent_, theTriggerResultsLabel;
+  edm::EDGetTokenT<trigger::TriggerEvent>  tok_trigEvt;
+  edm::EDGetTokenT<edm::TriggerResults>    tok_trigRes;
+
+  edm::EDGetTokenT<reco::TrackCollection>  tok_genTrack_;
+  edm::EDGetTokenT<reco::VertexCollection> tok_recVtx_;
+  edm::EDGetTokenT<reco::BeamSpot>         tok_bs_;
+  edm::EDGetTokenT<EcalRecHitCollection>   tok_EB_;
+  edm::EDGetTokenT<EcalRecHitCollection>   tok_EE_;
+  edm::EDGetTokenT<HBHERecHitCollection>   tok_hbhe_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+AlCaIsoTracksFilter::AlCaIsoTracksFilter(const edm::ParameterSet& iConfig) :
+  nRun(0), nAll(0), nGood(0) {
+  //now do what ever initialization is needed
+  trigNames                           = iConfig.getParameter<std::vector<std::string> >("Triggers");
+  theTrackQuality                     = iConfig.getParameter<std::string>("TrackQuality");
+  processName                         = iConfig.getParameter<std::string>("ProcessName");
+  l1Filter                            = iConfig.getParameter<std::string>("L1Filter");
+  l2Filter                            = iConfig.getParameter<std::string>("L2Filter");
+  l3Filter                            = iConfig.getParameter<std::string>("L3Filter");
+  reco::TrackBase::TrackQuality trackQuality_=reco::TrackBase::qualityByName(theTrackQuality);
```

Comment:
A minor point: trackQuality_ is a local variable, but its name ends with an underbar, which usually indicates the variable is a data member. You use this name in two more places in this file. A name like trackQualityTmp might be better.